rocky-raft/src/rocky/raft/server/CandidateLogic.java

```diff
@@ -32,7 +32,12 @@
         LOG_TAG += serverContext.getId();
         this.timeoutListener = timeoutListener;
         this.onMajorityReachedListener = onMajorityReachedListener;
-        this.clusterSize = Config.SERVERS.size();
+        try{
+            this.clusterSize = Config.getServers(serverContext.getLog()).size(); // TODO Needs to be updated on Configuration change?
```

Comment:
I think it'll be much cleaner if 

class Config {
   Config get() {
   // Returns immutable copy of current Config
   // Direct from log (& cached)
   }
}

class ServerConfig {
   int id;
   Address address;
}

So all ServerLogic can be passed with a copy of Config object when created. And when they receive update config message, they can tell the raft server to re-init them once config is written to the log.